src/wasm/wasm-type.cpp

```diff
@@ -433,15 +438,17 @@ std::optional<HeapType> getBasicHeapTypeLUB(HeapType::BasicHeapType a,
       }
       break;
     case HeapType::array:
-    case HeapType::string:
       lubUnshared = HeapType::any;
       break;
+    case HeapType::string:
     case HeapType::none:
     case HeapType::noext:
     case HeapType::nofunc:
     case HeapType::nocont:
     case HeapType::noexn:
-      // Bottom types already handled.
+      // Bottom types already handled (including string as the switch value,
+      // which implies the other value is bottom, which again would have been
+      // handled).
```

Comment:
I don't understand the addition to this comment. `string` isn't a bottom type, so why is it included in the bottom types already having been handled?

Reply:
Lines 399-400 swap to get the lower type as `a`. If `a` is string, then `b` is higher than `a`, which means (since string is the last of the non-bottom types in the list) that `b` is a bottom type. But we already handled bottom types before on 392-396.

Reply:
Right, but I don't get that from reading the comment. Is there something special about string here that doesn't also apply to other types?

Reply:
Just that strings are at the end (right before all the bottom types). I can add that to the comment.

Reply:
Comment improved.